lib/fake_ifdh.py

```diff
@@ -50,10 +50,22 @@ def getExp() -> Union[str, None]:
 
 def getRole(role_override: Optional[str] = None) -> str:
     """get current role"""
+
     if role_override:
         return role_override
-    if os.environ["USER"][-3:] == "pro":
-        return "Production"
+
+    # if there's a role in the wlcg.groups of the token, pick that
+    if os.environ.get("BEARER_TOKEN_FILE", False):
+        with os.popen("decode_token.sh $BEARER_TOKEN_FILE", "r") as f:
+            token_s = f.read()
+            token = json.loads(token_s)
+            groups: List[str] = token["wlcg.groups"]
+            if len(groups) == 2:
+                pos = groups[1].find("/", 2)
```

Comment:
As we discussed on slack, it might be more robust to make this a regex capturing group.  We know that the groups will look something like this:

```
    "/dune",
    "/dune/production"
```
So rather than assume that the production (or non-default role) will always come second, check each key against a regex; roughly:

```python
experiment = getExp()
role_regex = re.compile(f"\/{experiment}\/(\w+)"))
check_for_subgroup_match()
return subgroup_match
```